tests/ibr/test_service.py

```diff
@@ -0,0 +1,262 @@
+
+        res = run_coroutine(self.s.get_client_info())
+
+        self.cc.send.assert_called_once_with(unittest.mock.ANY)
+
+        _, (iq, ), _ = self.cc.send.mock_calls[0]
+
+        self.assertIsInstance(
+            iq,
+            aioxmpp.IQ,
+        )
+
+        self.assertEqual(
+            iq.to,
+            TEST_PEER.bare().replace(localpart=None),
+        )
+
+        self.assertEqual(
+            iq.type_,
+            aioxmpp.IQType.GET,
+        )
+
+        self.assertIsInstance(
+            iq.payload,
+            ibr_xso.Query,
+        )
+
+        self.assertIsInstance(
+            res,
+            ibr_xso.Query,
+        )
+
+        self.assertEqual(
+            res.username,
+            TEST_PEER.localpart,
+        )
+
+    def test_ibr_change_pass(self):
+        self.cc.local_jid = TEST_PEER
+        new_pass = "aaa"
+
+        run_coroutine(
+            self.s.change_pass(new_pass)
+        )
+
+        self.cc.send.assert_called_once_with(unittest.mock.ANY)
+
+        _, (iq, ), _ = self.cc.send.mock_calls[0]
+
+        self.assertIsInstance(
+            iq,
+            aioxmpp.IQ,
+        )
+
+        self.assertEqual(
+            iq.type_,
+            aioxmpp.IQType.SET,
+        )
+
+        self.assertEqual(
+            iq.to,
+            TEST_PEER.bare().replace(localpart=None),
+        )
+
+        self.assertIsInstance(
+            iq.payload,
+            ibr_xso.Query,
+        )
+
+        self.assertEqual(
+            iq.payload.username,
+            TEST_PEER.localpart,
+        )
+
+        self.assertEqual(
+            iq.payload.password,
+            new_pass,
+        )
+
+    def test_ibr_cancel_registration(self):
+        self.cc.local_jid = TEST_PEER
+
+        run_coroutine(self.s.cancel_registration())
+
+        self.cc.send.assert_called_once_with(unittest.mock.ANY)
+
+        _, (iq, ), _ = self.cc.send.mock_calls[0]
+
+        self.assertIsInstance(
+            iq,
+            aioxmpp.IQ,
+        )
+
+        self.assertEqual(
+            iq.to,
+            TEST_PEER.bare().replace(localpart=None),
+        )
+
+        self.assertEqual(
+            iq.type_,
+            aioxmpp.IQType.SET,
+        )
+
+        self.assertIsInstance(
+            iq.payload,
+            ibr_xso.Query,
+        )
+
+        self.assertEqual(
+            iq.payload.remove,
+            True,
+        )
+
+    def test_ibr_get_registration_fields(self):
+        with unittest.mock.patch(
+                'aioxmpp.protocol.send_and_wait_for',
+                new=CoroutineMock()
+        ) as mock1:
+            iq_res = aioxmpp.IQ(
+                type_=aioxmpp.IQType.GET,
+                payload=ibr_xso.Query()
+            )
+            iq_res.payload.username = ''
+            mock1.return_value = iq_res
+            stream = aioxmpp.protocol.XMLStream(
+                to=TEST_PEER.domain,
+                features_future=asyncio.Future()
+            )
```

Comment:
Instead of creating the stream here, use a mock object:

```python
stream = unittest.mock.Mock(spec=aioxmpp.protocol.XMLStream)
stream._to = TEST_PEER.domain
```